src/metabase/models/action.clj

```diff
@@ -209,3 +214,160 @@
                         (m/index-by :id (actions-with-implicit-params (map :card dashcards) :id [:in action-ids])))]
     (for [dashcard dashcards]
       (m/assoc-some dashcard :action (get actions-by-id (:action_id dashcard))))))
+
+;;; ------------------------------------------------ Serialization ---------------------------------------------------
+
+(defmethod serdes.hash/identity-hash-fields Action
+  [_action]
+  [:name (serdes.hash/hydrated-hash :model "<none>") :created_at])
+
+(defmethod serdes.hash/identity-hash-fields QueryAction
+  [_action]
+  [(serdes.hash/hydrated-hash :action "<none>")])
+
+(defmethod serdes.hash/identity-hash-fields ImplicitAction
+  [_action]
+  [(serdes.hash/hydrated-hash :action "<none>")])
+
+(defmethod serdes.hash/identity-hash-fields HTTPAction
+  [_action]
+  [(serdes.hash/hydrated-hash :action "<none>")])
+
+(defmethod serdes.base/extract-one "Action"
+  [_model-name _opts action]
+  (-> (serdes.base/extract-one-basics "Action" action)
+      (update :creator_id serdes.util/export-user)
+      (update :model_id serdes.util/export-fk 'Card)))
+
+(defmethod serdes.base/load-xform "Action" [action]
+  (-> action
+      serdes.base/load-xform-basics
+      (update :model_id serdes.util/import-fk 'Card)
+      (update :creator_id serdes.util/import-fk-keyed 'User :email)))
+
+(defmethod serdes.base/extract-one "QueryAction"
+  [_model-name _opts query-action]
+  (-> (serdes.base/extract-one-basics "QueryAction" query-action)
+      (update :database_id serdes.util/export-fk-keyed 'Database :name)
+      (assoc :action_id (serdes.util/export-fk (:action_id query-action) 'Action))))
```

Comment:
There's no `:action_id` in the threaded value because action_id is the primary key, which is removed by `extract-one-basics` 
See
https://github.com/metabase/metabase/blob/933c702176d75ccde0feb4d209f4176e371f932b/src/metabase/models/serialization/base.clj#L262

Reply:
Not relevant anymore, since QueryActions are inlined